### ISSUE TITLE
Skip network tests when network is unavailable

### ISSUE THREAD

**Issue by talex5** [ocaml-multicore/eio#275]
The network tests can't be run in some cases:

- Docker containers don't have IPv6 loopback by default.
- opam-repo-ci's sandbox on macos prevents using any loopback devices.

This is probably best done by extending MDX. We could perhaps have an `exception Mdx_skip_block` to indicate that the current block's output should be ignored, or a `skip-if=TEST` label at the top of the block.

From #274.

**Comment by talex5** [comment]
Yes. This would be a good first step: https://github.com/realworldocaml/mdx/issues/393

**Comment by create2000** [comment]
Thank you @talex5 -- I'll have to wait for @patricoferris to assign me the issue.

**Comment by patricoferris** [comment]
Assigned :))

**Comment by talex5** [comment]
To get started, you can try running the tests under Docker, using the Dockerfile in the repository. e.g.

```
docker build . -t eio
docker run -it --rm eio
opam install --deps-only -t eio_main eio_linux eio_posix eio
make
```
 
I just tried this, and it seems that Docker does provide IPv6 now (at least on the loopback interface). However, it now blocks uring:

```
$ dune runtest ./lib_eio_linux/tests/
...
@@ -133,6 +129,5 @@ Signalling an exited child does nothing:
   in
   ignore (Promise.await (Process.exit_status child) : Unix.process_status);
   Process.signal child Sys.sigkill;;
-FOO=bar
-- : unit = ()
+Exception: Failure "io_uring is not available (permission denied)".
 ```

So probably the tests will need changing to `Eio_linux.run ~fallback:skip`, where `skip` raises some `Mdx_skip` exception.

Note: this is marked "good first issue" as it doesn't require much knowledge of Eio and the changes needed here are simple. However, adding the skip feature to MDX might be more difficult!

**Comment by create2000** [comment]
Hello @patricoferris -- I am making my Outreachy final application and I am asked to create a project timeline. Following the Project details, I came up with a timeline. I want to share it with you so if there's any correction before I submit, or something you want me to insert or remove. Here's the link: https://docs.google.com/document/d/108a5nU0J9Z2tFIUR09lwPmwww4796qpFVx3nZNSKilc/edit?usp=drivesdk
